nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/rs2d/RS2DData.java

```diff
@@ -1241,6 +1245,25 @@ public void setHeaderPhases(Dataset dataset) throws XPathExpressionException {
         header.<ListNumberValue>get(PHASE_1).setValue(phase1Values);
     }
 
+    public void setHeaderState(Dataset dataset) {
+        if (!header.contains(STATE)) {
+            header.put(new ListNumberValue(STATE.name(), Collections.emptyList()));
+            ListNumberValue stateParam = header.get(STATE);
+            stateParam.setUuid(UUID.randomUUID().toString());
+            stateParam.setNumberEnum(NumberType.Integer);
+            stateParam.setMinValue(Integer.MIN_VALUE);
+            stateParam.setMaxValue(Integer.MAX_VALUE);
+            Order order = Stream.of(Order.values()).filter(o -> o.toString().equals(dataset.getNDim() + "D")).findFirst().orElse(Order.Unknown);
+            stateParam.setOrder(order);
+        }
+        List<Number> stateValues = new ArrayList<>(Collections.nCopies(4, 0));
+        for (int i = 0; i< dataset.getNDim(); i++) {
+            stateValues.set(i, 1);
```

Comment:
This isn't really correct. It should contain 1 in dimensions where a fourier transform was done, and 0 in the others. I'm not sure if the dataset contains this information? Otherwise we'll keep your approximation.

Reply:
One example is pseudo 2D (T1, T2, ...), where FT is done in the first dimension only.